.github/ISSUE_TEMPLATE/07-optimization.yml

```diff
@@ -0,0 +1,72 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+name: "Internal: proposed optimization"
+description: A proposal for a new query optimization. Contributor use only.
+labels: ["A-optimization", "C-feature", "needs-discussion"]
+body:
+  - type: textarea
+    attributes:
+      label: Elevator pitch
+      description: |
+        Each question should be answered comprehensively but using as few sentences as possible. This is the only mandatory section needed for the issue to be discussed in a refinement session.
+      value: |
+        #### What does the transformation do?
+        <!-- Give a representative SQL query. -->
+
+        #### Why should we add it?
+
+        #### When would it be good to have?
+
+        #### When would it be ineffectual?
+
+        #### When would be bad to have?
+
+        #### In the worst case, how would it degrade performance?
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Customer impact
+      description: |
+          List real life instances where this transformation would help.
+  - type: textarea
+    attributes:
+      label: Cost model
+      description: |
+        If both the implementation and cost model are simple, this section can be skipped if the implementation section is filled in.
+
+        Each subsection is meant to be answered with formulae. Take into account memory, CPU, liveness, and skew. Approximations are ok if it is not possible to calculate exact numbers.
```

Comment:
```suggestion
        Take into account memory, CPU, liveness, and skew. If possible, each subsection can be answered with formulae. Approximations are ok, too.
```

Reply:
The cost model is an approximation by definition, so I am not sure why we need to discuss anything about approximation.

I think we agreed that should have a cost model - not necessary before refinement, but before the optimization is signed-off and we can start working on the implementation. How good an approximation this cost model is w.r.t. the real world needs to be judged case by case.

Reply:
I updated my suggestion to remove `Approximations are ok, too.`